components/event-bus/internal/knative/subscription/controller/eventactivation/reconcile.go

```diff
@@ -3,112 +3,123 @@ package eventactivation
-		log.Info("EventActivation reconciled")
-		r.recorder.Eventf(ea, corev1.EventTypeNormal, "EventactivationReconciled", "EventActivation reconciled, name: %q; namespace: %q", ea.Name, ea.Namespace)
+		r.Recorder.Eventf(ea, corev1.EventTypeNormal, eventactivationreconciled, "EventActivation reconciled, name: %q; namespace: %q", ea.Name, ea.Namespace)
 	}
-	return reconcile.Result{
-		Requeue: requeue,
-	}, reconcileErr
+
+	return reconcileErr
 }
 
-func (r *reconciler) reconcile(ctx context.Context, ea *eventingv1alpha1.EventActivation) (bool, error) {
+// eventActivationByKey retrieves a EventActivation object from a lister by ns/name key.
+func eventActivationByKey(key string, l applicationconnectorlistersv1alpha1.EventActivationLister) (*applicationconnectorv1alpha1.EventActivation, error) {
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "invalid object key"))
+	}
+
+	src, err := l.EventActivations(ns).Get(name)
+	switch {
+	case apierrors.IsNotFound(err):
+		return nil, controller.NewPermanentError(pkgerrors.Wrap(err, "object no longer exists"))
+	case err != nil:
+		return nil, err
+	}
+
+	return src, nil
+}
 
+func (r *Reconciler) reconcile(ctx context.Context, ea *applicationconnectorv1alpha1.EventActivation) (bool, error) {
+	log := logging.FromContext(ctx)
 	// delete or add finalizers
 	if !ea.DeletionTimestamp.IsZero() {
 		// deactivate all Kyma subscriptions related to this ea
-		subs, _ := util.GetSubscriptionsForEventActivation(ctx, r.client, ea)
-		util.DeactivateSubscriptions(ctx, r.client, subs, log, r.time)
+		subs, _ := util.GetSubscriptionsForEventActivation(r.kymaEventingClient, ea)
+		util.DeactivateSubscriptions(r.kymaEventingClient, subs, log, r.time)
 
 		// remove the finalizer from the list
 		ea.ObjectMeta.Finalizers = util.RemoveString(&ea.ObjectMeta.Finalizers, finalizerName)
-		log.Info("Finalizer removed", "Finalizer name", finalizerName)
+		log.Info("Finalizer removed", zap.String("Finalizer name", finalizerName))
 		return false, nil
 	}
 
 	// If we are adding the finalizer for the first time, then ensure that finalizer is persisted
 	if !util.ContainsString(&ea.ObjectMeta.Finalizers, finalizerName) {
 		//Finalizer is not added, let's add it
 		ea.ObjectMeta.Finalizers = append(ea.ObjectMeta.Finalizers, finalizerName)
-		log.Info("Finalizer added", "Finalizer name", finalizerName)
+		log.Info("Finalizer added", zap.String("Finalizer name", finalizerName))
 		return true, nil
 	}
 
-	// check an activate, if necessary, all the subscriptions
-	if subs, err := util.GetSubscriptionsForEventActivation(ctx, r.client, ea); err != nil {
-		log.Error(err, "GetSubscriptionsForEventActivation() failed")
+	// check and activate, if necessary, all the subscriptions
+	if subs, err := util.GetSubscriptionsForEventActivation(r.kymaEventingClient, ea); err != nil {
+		log.Error("GetSubscriptionsForEventActivation() failed", zap.Error(err))
```

Comment:
Why not return `false, err` after this line?

Reply:
Makes sense, however, this was the original state.

Reply:
It makes no sense to return a boolean. Knative reconcilers only evaluate the returned error. `nil` means that the reconciliation was successful. `Permanent` error means "don't retry". Any other error is retried.

Reply:
Removed the bool return